### ISSUE TITLE
Move mapbox key to settings.json

### ISSUE THREAD

**Issue by mollybsmith-noaa** [NOAA-GSL/MATS#567]
---

Author Name: **molly.b.smith** (@mollybsmith-noaa)
Original Redmine Issue: 81863, https://vlab.ncep.noaa.gov/redmine/issues/81863
Original Date: 2020-08-12
Original Assignee: molly.b.smith

---

This needs to happen before MATS 2.6.1 goes out next week with Dave's stdev change.

**Comment by mollybsmith-noaa** [comment]
---

Original Redmine Comment
Author Name: **molly.b.smith** (@mollybsmith-noaa)
Original Date: 2020-08-15T00:15:57Z

---

I have moved the MATS mapbox key out of the app code and into each app’s settings.json, which is not part of the MATS git repo (although we do store our settings files in a different repo to keep track of them). Once MATS 2.6.1 has eventually propagated to all of our servers, I will disable our current key (which is in the git history) and generate a new one.

If there is a private.MAPBOX_KEY in an app’s settings file, it will act indistinguishably from how it did before. If that field does not exist, or if it is the string “undefined” (which I think should be the default value on setup, unless a user supplies a key when configuring the app for the first time), the user is presented with the error message "The mapbox access token is currently undefined, so the sitesMap selector and map plots will not work as expected. To fix this, create an account at mapbox.com, generate a free access token, and add it to your settings.json file as private.MAPBOX_KEY on page load". If the user subsequently ignores this and still tries to generate a map plot, they receive the error message "[Error in getGraphData function:dataMap : The mapbox access token is currently undefined, so MATS cannot produce a map plot at this time. To fix this, create an account at mapbox.com, generate a free access token, and add it to your settings.json file as private.MAPBOX_KEY.]" which I hope will finally get the point across.

I believe I have done everything that needs to be done with this, except modifying the setup script, because I didn’t want to mess up any of Randy’s kubernetes work. It should ask the user to supply a mapbox key, and if they don’t have one, it should set private.MAPBOX_KEY to “undefined". 

If you are running MATS locally, you will need to change your own settings.json files to include this new field.
